File: request_app/lib/models/reassignment.dart

import 'package:request_app/models/request.dart';
import 'package:request_app/models/user.dart';

class Reassignment {
  const Reassignment(
    this.request,
    this.reassignedFrom,
    this.reassignedTo,
    this.reassignedAt,
  );

  final Request request;
  final User reassignedFrom;
  final User reassignedTo;
  final DateTime reassignedAt;
}

File: server/models/reassignment.js
import mongoose from "mongoose";
import Request from "./request.js";

const reassignmentSchema = new mongoose.Schema({
    request: {
        type: mongoose.Schema.Types.ObjectId,
        ref: "Request",
        required: true
    },
    reassignedFrom: {
        type: mongoose.Schema.Types.ObjectId,
        ref: "User",
        required: true
    },
    reassignedTo: {
        type: mongoose.Schema.Types.ObjectId,
        ref: "User",
        required: true
    },
    reassignedAt: {
        type: Date,
        default: Date.now
    }
});

const Reassignment = mongoose.model("Reassignment", reassignmentSchema);

export default Reassignment;